website/docs/reference/appsmith-framework/query-object.md

```diff
@@ -3,91 +3,119 @@ sidebar_position: 3
+
+On the Checkbox-group's `selectedValues` property, there's an array of the selected values.
 
 ```javascript
-{{ this.params.key }}
+// CheckboxGroup1.selectedValues
+[ "1", "3", "4", "6", "10" ]
 ```
 
-Please see the quick demo below showing how to access your params from `Query.run(params)` within the query. We'll use the [Postman Echo API](https://learning.postman.com/docs/developer/echo-api/), which will echo our request. We'll pass it `{ phrase: "hello, world!" }` , and then receive it back as a response:
-
+Next, set up a helper function `queryTeams` in a [JS Object](/core-concepts/writing-code/javascript-editor-beta) called `utils`:
 
-<VideoEmbed host="youtube" videoId="oktXirbay8U" title="Use this.params.key within your query to access any params you passed within the .run(params) function." caption="Use this.params.key within your query to access any params you passed within the .run(params) function."/>
+```javascript
+// function in a JS Object
+  queryTeams: async (teamsArray) => {
+    const teamsPromises = await teamsArray.map(team => {
+      GetTeamsData.run({ "team": team })
+    })
+    const responseArray = await Promise.allSettled(teamsPromises);
+    const result = responseArray.map( result => result.value? )
+    await storeValue("teamsData", result)
+  }
+```
 
-### onSuccess
+The `queryTeams` function takes the user's selection from the CheckboxGroup, and runs the query once for each selected team, substituting the team number into the query. Once all the queries have completed, the function creates an array of their aggregated data from the responses and stores them in the [Appsmith Store](/reference/appsmith-framework/widget-actions/store-value).
 
-The onSuccess function is run when a query runs successfully. The function returns the response of the query and the params passed to it in the callback arguments.
+Back on the canvas, call your function to run the queries in a [Button widget's](/reference/widgets/button) **onClick** field:
 
 ```javascript
-onSuccess(response, params): void
+// Button widget's onClick
+{{ utils.queryTeams(CheckboxGroup1.selectedValues) }}
 ```
 
-### onError
-
-The onError function is run when a query execution fails. The function returns the response of the query and the params passed to it.
+Create a [Table widget](/reference/widgets/table) and bind your stored data into the **Table Data** field:
 
 ```javascript
-onError(response, params): void
+{{ appsmith.store.teamsData }}
 ```
 
-## Data
+Now you can select your teams with the checkboxes, click the button, and see your table populate with data.
 
-Each query stores the data from its latest run inside its **data** property. This property is populated only if the query successfully executes and can be accessed as
+### Callbacks (deprecated)
```

Comment:
I'm hesitant to add code snippets for all the different async styles, since it's not really related to the query object or to appsmith; the "Callbacks" heading just exists since it's a different function signature that requires using the `run()` function in a different way.

I'll add them though to see what it looks like, because I feel like it's making a bigger deal in my head than it actually is 🙃

Reply:
after review, "Callbacks" style was removed entirely, 2 async styles added via Tabs, `run()` params example replaced with a simpler version that is actually testable by users if they wish via the mock Postgres db